packages/server-boilerplate/src/connections/ApiConnectionBuilder.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ *
+ */
+import { MicroserviceApi, AuthHandler } from './types';
+import { ApiConnection } from './ApiConnection';
+
+export class ApiConnectionBuilder {
+  private authHandler?: AuthHandler;
+
+  public handleAuthWith(authHandler: AuthHandler) {
+    this.authHandler = authHandler;
+    return this;
+  }
+
+  public buildAs<Wrapper extends MicroserviceApi>(
+    WrapperClazz: new (apiConnection: ApiConnection) => Wrapper,
+  ) {
+    if (!this.authHandler) {
+      throw new Error('Must specify an authHandler when building an ApiConnection');
+    }
+
+    const apiConnection = new ApiConnection(this.authHandler);
+    const wrappedApiConnection = new WrapperClazz(apiConnection);
+    apiConnection.baseUrl = wrappedApiConnection.baseUrl;
+    return wrappedApiConnection;
```

Comment:
This little chunk here is super hacky, and as soon as we get a chance to move all ApiConnections over to using this builder pattern I'd love to change it... I feel ideally there'd be a static baseUrl property/function on all wrapper classes that we could use to instantiate the ApiConnection instance with?